frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -645,60 +645,87 @@ export function ContainerInfo(props: ContainerInfoProps) {
   const { container, status, resource } = props;
   const { t } = useTranslation(['glossary', 'translation']);
 
-  const [startedDate, finishDate] = React.useMemo(() => {
-    let startedDate =
-      status?.state?.running?.startedAt || status?.state?.terminated?.startedAt || '';
-    if (!!startedDate) {
-      startedDate = localeDate(startedDate);
+  const [startedDate, finishDate, lastStateStartedDate, lastStateFinishDate] = React.useMemo(() => {
+    function getStartedDate(state?: KubeContainerStatus['state']) {
+      let startedDate = state?.running?.startedAt || state?.terminated?.startedAt || '';
+      if (!!startedDate) {
+        startedDate = localeDate(startedDate);
+      }
+      return startedDate;
     }
 
-    let finishDate = status?.state?.terminated?.finishedAt ?? '';
-    if (!!finishDate) {
-      finishDate = localeDate(finishDate);
+    function getFinishDate(state?: KubeContainerStatus['state']) {
+      let finishDate = state?.terminated?.finishedAt || '';
+      if (!!finishDate) {
+        finishDate = localeDate(finishDate);
+      }
+      return finishDate;
     }
 
-    return [startedDate, finishDate];
+    return [
+      getStartedDate(status?.state),
+      getFinishDate(status?.state),
+      getStartedDate(status?.lastState),
+      getFinishDate(status?.lastState),
+    ];
   }, [status]);
 
-  function getContainerStatusLabel() {
-    if (!status || !container) {
-      return undefined;
-    }
+  function ContainerStatusLabel(props: {
+    state?: KubeContainerStatus['state'] | null;
+    container?: KubeContainer;
+  }) {
+    const { state, container } = props;
+
+    const [stateDetails, label, statusType] = React.useMemo(() => {
```

Comment:
also here useMemo is not necessary

Reply:
Well, this one uses translation and all. I know it's not a lot of gains, but I feel like we should be processing all this every time there's a slight update.

Reply:
yeah I'm not sure of the `t` function call cost. in theory we can move it out of component somewhere at the top level but it doesn't matter that much